src/main/java/sigma/Storage.java

```diff
@@ -0,0 +1,153 @@
+     */
+    private void load(String filePath) {
+        // Creates a file to store data if none exists
+        try {
+            Files.createFile(Paths.get(filePath));
+        } catch (IOException e) {
+            System.out.println("SLAY! Loading up your saved tasks!");
+        }
+    }
+
+    /**
+     * Reads the tasks from the file.
+     * Creates a list of tasks from the file.
+     * Handles the reading of tasks from the file.
+     * Handles the case where the file is not found.
+     * Handles the case where the file is empty.
+     * @param data File to read from.
+     * @return List of tasks read from the file.
+     */
+    public ArrayList<Task> read(File data) {
+        ArrayList<Task> items = null;
+        try {
+            items = new ArrayList<>();
+            Scanner fileInput = new Scanner(data);
+            while (fileInput.hasNextLine()) {
+                // Loading from file
+                String line = fileInput.nextLine();
+                String[] split = line.split(" \\| ");
+                String type = split[0];
+                boolean status = split[1].equals("X");
+                String desc = split[2];
+                String date = split[3];
+                Task item = createTaskFromFile(type, desc, date);
+                item.setStatus(status);
+                items.add(item);
+            }
+            fileInput.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred while reading file.");
+            e.printStackTrace();
+        } catch (SigmaException e) {
+            System.out.println(e.getMessage());
+        }
+        return items;
+    }
+
+    /**
+     * Creates a task from the file.
+     * @param type Task type.
+     * @param desc Task description.
+     * @param date Task date.
+     * @return Task created from the file.
+     * @throws SigmaException If the task type is invalid, or contains invalid date format.
+     */
+    private Task createTaskFromFile(String type, String desc, String date) throws SigmaException {
+        Task item;
+        switch (type) {
+        case "T":
+            item = new ToDoTask(desc);
+            break;
+        case "D":
+            LocalDateTime dateTime;
+            try {
+                dateTime = LocalDateTime.parse(date.strip(), DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm"));
+            } catch (DateTimeParseException e) {
+                throw new SigmaException("What the sigma? File contains invalid date format!");
+            }
+            item = new DeadlineTask(desc, dateTime);
+            break;
+        case "E":
+            String[] dates = date.split("-");
+            LocalDateTime from;
+            LocalDateTime to;
+            try {
+                from = LocalDateTime.parse(dates[0].strip(), DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm"));
+                to = LocalDateTime.parse(dates[1].strip(), DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm"));
```

Comment:
Perhaps declare `DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm")` as a constant? Then you don't need to change both lines if the date format changes in the future.

Reply:
Will do!